src/devices/Ina226/Ina226.cs

```diff
@@ -0,0 +1,489 @@
+        {
+            ushort regValue = ReadRegister(Ina226Register.Current, _shuntConvTimeInt * _samplesAveragedInt);
+            if ((regValue & 0x8000) > 0)
+            {
+                regValue &= 0x7FFF;
+                regValue ^= 0x7FFF;
+                regValue++;
+
+                return ElectricCurrent.FromAmperes(regValue * -_currentLsb);
+            }
+
+            return ElectricCurrent.FromAmperes(regValue * _currentLsb);
+        }
+
+        /// <summary>
+        /// Get the calculated power of the circuit being monitored by the INA226
+        /// </summary>
+        /// <remarks>
+        /// This value is determined by an internal calculation using the calculated current and the read bus voltage and then scaled.
+        /// </remarks>
+        /// <returns>The calculated electric power</returns>
+        [Telemetry("Power")]
+        public Power ReadPower()
+        {
+            return Power.FromWatts(ReadRegister(Ina226Register.Power, _shuntConvTimeInt * _samplesAveragedInt) * _currentLsb * 25);
+        }
+
+        /// <summary>
+        /// Get the status of the AlertFunctionFlag bit on the INA226
+        /// </summary>
+        /// <returns>
+        /// True if flag is set and false if not
+        /// </returns>
+        [Telemetry("AlertFunctionFlag")]
+        public bool AlertFunctionFlag()
+        {
+            return IsBitSet((byte)(ReadRegister(Ina226Register.AlertEnable) & (ushort)(Ina226AlertMask.AlertFunctionFlag)), 4);
+        }
+
+        /// <summary>
+        /// Get the status of the Conversion Ready Flag bit on the INA226
+        /// </summary>
+        /// <returns>
+        /// True if flag is set and false if not
+        /// </returns>
+        [Telemetry("ConversionReadyFlag")]
+        public bool ConversionReadyFlag()
+        {
+            return IsBitSet((byte)(ReadRegister(Ina226Register.AlertEnable) & (ushort)(Ina226AlertMask.AlertConvReadyFlag)), 3);
+        }
+
+        /// <summary>
+        /// Get the status of the Math Overflow Flag bit on the INA226
+        /// </summary>
+        /// <returns>
+        /// True if flag is set and false if not
+        /// </returns>
+        [Telemetry("MathOverflowFlag")]
+        public bool MathOverflowFlag()
+        {
+            return IsBitSet((byte)(ReadRegister(Ina226Register.AlertEnable) & (ushort)(Ina226AlertMask.AlertMathOverflowFlag)), 2);
+        }
+
+        /// <summary>
+        /// Property representing the Operating mode of the INA226
+        /// </summary>
+        /// <remarks>
+        /// This allows the user to select continuous, triggered, or power-down mode of operation along with which of the shunt and bus voltage measurements are made.
+        /// </remarks>
+        [Property]
+        public Ina226OperatingMode OperatingMode
+        {
+            get => (Ina226OperatingMode)(ReadRegister(Ina226Register.Configuration) & (ushort)Ina226ConfigurationFlags.ModeMask);
```

Comment:
Properties should not normally access hardware directly, particularly in getters. Preferably use a cached value instead.

Reply:
So a more proper way to do this would be to cache the values when set and then only retrieve them from cached values if i am understanding you correctly?

Reply:
Yes. That would be better, because it's significantly faster and reduces risks of side effects. Particularly when debugging, this may otherwise cause unexpected behavior, because the debugger evaluates the properties as well, which then causes a hardware access (eventually with the device not in a state ready for it)

Reply:
Ah that makes complete sense. I didn't think about the debugger accessing the hardware eventuality.